### PR TITLE
sdk consistent services

### PR DESCRIPTION
**Pull request by joshbax189**
Changes:
- All SDK services inherit from `Service<API>` that provides keystore/vault API factory and loggers.
  The API type specifies where any extra endpoints a user would look for are located, e.g. `ConnectionService extends Service<ConnectionAPI>` and so we can do `ConnectionService.getAPI(token).delete()`. Note that services are not limited to only this endpoint/API, they can still create other API instances via the API factories.
- All service methods take API credentials as their first parameter and always as an object.
- API credentials are tagged: e.g. `interface IVaultToken = { vault_access_token: string }`, so you cannot pass/get a vault token instead of a keystore token by mistake.
- Tagged credentials are used in method signatures: `getItem(credentials: IVaultToken & IDEK, id: string): Item`; You can provide just what is needed, or pass an AuthData object for credentials.
- Replaced most instances of encryption/decryption of slots with `ItemService(encrypt|decrypt)Slot`.
- Moved `ConnectionCreateData` to `ConnectionService` and make it an interface
- Moved `fetchConnectionById` etc to `ConnectionService`
- Moved `SecretService` to `util/Secrets`.
- ~~Bump SDK major version~~ TODO

### CODE REVIEW

File: packages/cli/package.json [hunk 1]
```diff
@@ -12,7 +12,7 @@
     "@meeco/cryppo": "^0.9.0",
     "@meeco/file-storage-node": "2.1.1",
     "@meeco/keystore-api-sdk": "^0.14.0",
-    "@meeco/sdk": "^1.0.0",
+    "@meeco/sdk": "^2.0.0",
```
Comment (spazworm) [inline packages/cli/package.json:15]: probably we can have this be 2.0.0-beta until we've done some testing

File: packages/sdk/src/services/invitation-service.ts [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+      key_encryption_key,
+      data_encryption_key,
+    } = credentials;
+    const keyPair = await this.createAndStoreKeyPair(keystore_access_token, key_encryption_key);
+
+    this.logger.log('Encrypting connection name');
+    const encryptedName: string = await this.encryptName(name, data_encryption_key);
+
+    this.logger.log('Accepting invitation');
+    return await this.vaultAPIFactory(vault_access_token)
```
Comment (zbarbuto) [inline packages/sdk/src/services/invitation-service.ts:57]: My preference is to not `return await` unless you actually need to (i.e. if it's in a try-catch block so you can handle the error). You might as well just return the promise.

File: packages/sdk/src/services/item-service.ts [hunk 3]
```diff
@@ -121,103 +129,136 @@ export class ItemService {
...
         slots_attributes,
       },
     });
   }
 
-  public async removeSlot(slotId: string, vaultAccessToken: string) {
+  public async removeSlot(credentials: IVaultToken, slotId: string) {
     this.logger.log('Removing slot');
-    await this.vaultAPIFactory(vaultAccessToken).SlotApi.slotsIdDelete(slotId);
+    await this.vaultAPIFactory(credentials.vault_access_token).SlotApi.slotsIdDelete(slotId);
```
Comment (zbarbuto) [inline packages/sdk/src/services/item-service.ts:149]: Unless I'm just getting confused mixing `this.vaultAPIFactory` and `this.getAPI` feels a bit confusing to me?

File: packages/sdk/src/services/invitation-service.ts (hunk 2 above)
Comment (joshbax189) [reply packages/sdk/src/services/invitation-service.ts:57]: My preference too! Good spot!

File: packages/sdk/src/services/item-service.ts (hunk 3 above)
Comment (joshbax189) [reply packages/sdk/src/services/item-service.ts:149]: I made vault/keystoreAPIFactory protected in all services, but perhaps it's confusing mixing getAPI and APIFactory methods inside the class? Would it be less confusing if I only used APIFactory methods within the class say?

File: packages/cli/package.json (hunk 1 above)
Comment (joshbax189) [reply packages/cli/package.json]: No problem

File: packages/sdk/src/services/item-service.ts (hunk 3 above)
Comment (zbarbuto) [reply packages/sdk/src/services/item-service.ts:149]: Yeah I think I get that `getApi` is like a specific api (e.g. `SlotsApi`) but it's also a bit confusing to read through having both in there.

**Pull request merged**